### PR TITLE
58 client.js created. GET popular movies with axios

### PR DESCRIPTION
**Pull request by Ninolina88**
Client.js created and imported into the media.js file. Axios is functional and Popular movie's response is received in the terminal.

### CODE REVIEW

File: public/media-hunt/client.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const axios = require("axios");
+const instance = axios.create({
+  baseURL: "https://api.themoviedb.org/3/",
+  timeout: 2500,
+  params: {api_key: '109f8c8c06f943d5e337e7141b1dad67'}
+});
```
Comment (pvukmanovic-pannonians) [inline public/media-hunt/client.js:6]: This is incorrect. Please refer to postman or TMDB API documentation. Let's use a header, not param.

File: public/media-hunt/index.js [hunk 2]
```diff
@@ -1 +1,3 @@
-console.log("Media hunt");
\ No newline at end of file
+const app = require("./mediahunt.js");
+
+app.run();
```
Comment (pvukmanovic-pannonians) [inline public/media-hunt/index.js:3]: Please remove this. This file will be responsible for parsing node commands.

File: public/media-hunt/mediahunt.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    console.error(error);
+  }
+}
+
+async function run() {
+  const allMedia = await getData("movie/popular?language=en-US&page=1");
+  console.log(allMedia);
+}
+
+module.exports = { run };
```
Comment (pvukmanovic-pannonians) [inline public/media-hunt/mediahunt.js:19]: This entire file is incorrect. 
Please create separate files for movies and tv and conf., which will have their own api fetch methods.
Also, group them maybe in a folder called "http", and have your axios instance also generated there.

Example:
```
// movie.js

// Import instance and required path for movies from your ali scheme 

const getMovies = async () => {
  const { data } = await instance.get(movieUrl);
  return data;
}

...

module.exports = { getMovies, ... }
```

But also let's use "discover" API calls to get movies and tv series.

**Review by pvukmanovic-pannonians — CHANGES_REQUESTED** [review]

File: public/media-hunt/client.js (hunk 1 above)
Comment (Ninolina88) [reply public/media-hunt/client.js:6]: am i supposed to use bearer token here?
example: curl --request GET \
  --url 'https://api.themoviedb.org/3/movie/76341' \
  --header 'Authorization: Bearer <<access_token>>' \
  --header 'Content-Type: application/json;charset=utf-8'

Comment (Ninolina88) [reply public/media-hunt/client.js]: header with token implemented, please review

File: public/media-hunt/index.js (hunk 2 above)
Comment (Ninolina88) [reply public/media-hunt/index.js]: removed

File: public/media-hunt/client.js (hunk 1 above)
Comment (Ninolina88) [reply public/media-hunt/client.js]: today i will deal with the secret token, therefore this userstory is still not completed

**Review by pvukmanovic-pannonians — CHANGES_REQUESTED** [review]

File: public/media-hunt/client.js (hunk 1 above)
Comment (Ninolina88) [reply public/media-hunt/client.js]: NOTE: dotenv dependencie installed and env file placed in the root folder

File: public/media-hunt/client.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const axios = require("axios");
+
+const instance = axios.create({
+  baseURL: "https://api.themoviedb.org/3/",
+  timeout: 2500,
+});
+
+instance.default.headers.common = {
+  authorization:
+    "Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiJmMzEwZDg2NTM4OTkzYmNhMjNlNWUyOWNkYjk3NGFlNCIsInN1YiI6IjVjMWM5OTQwOTI1MTQxMjdmODRlMWVmNSIsInNjb3BlcyI6WyJh",
```
[the remark this answers is not in the document]
Comment (Ninolina88) [reply public/media-hunt/client.js]: done

File: public/media-hunt/index.js [hunk 5]
```diff
@@ -1 +1 @@
-console.log("Media hunt");
\ No newline at end of file
+const app = require("./mediahunt.js");
```
[the remark this answers is not in the document]
Comment (Ninolina88) [reply public/media-hunt/index.js:1]: done

File: public/media-hunt/API/movie.js [hunk 6]
```diff
@@ -1,4 +1,4 @@
-export default {
+module.export = {
```
Comment (pvukmanovic-pannonians) [inline public/media-hunt/API/movie.js:1]:
```suggestion
module.exports = {
```

File: public/media-hunt/API/tv.js [hunk 7]
```diff
@@ -1,4 +1,4 @@
-export default {
+module.export = {
```
Comment (pvukmanovic-pannonians) [inline public/media-hunt/API/tv.js:1]:
```suggestion
module.exports = {
```

File: public/media-hunt/API/configuration.js [hunk 8]
```diff
@@ -1,4 +1,4 @@
-export default {
+module.export = {
```
Comment (pvukmanovic-pannonians) [inline public/media-hunt/API/configuration.js:1]:
```suggestion
module.exports = {
```

File: public/media-hunt/API/index.js [hunk 9]
```diff
@@ -1,5 +1,5 @@
-import configuration from "./configuration";
-import movie from "./movie";
-import tv from "./tv";
+const { configuration } = require("./configuration");
```
Comment (pvukmanovic-pannonians) [inline public/media-hunt/API/index.js:1]:
```suggestion
const configuration = require("./configuration");
```

File: public/media-hunt/API/index.js [hunk 10]
```diff
@@ -1,5 +1,5 @@
...
+const { movie } = require("./movie");
```
Comment (pvukmanovic-pannonians) [inline public/media-hunt/API/index.js:2]:
```suggestion
const movie = require("./movie");
```

File: public/media-hunt/API/index.js [hunk 11]
```diff
@@ -1,5 +1,5 @@
...
+const { tv } = require("./tv");
```
Comment (pvukmanovic-pannonians) [inline public/media-hunt/API/index.js:3]:
```suggestion
const tv = require("./tv");
```

File: public/media-hunt/API/index.js [hunk 12]
```diff
@@ -1,5 +1,5 @@
...
 
-export { configuration, movie, tv };
+module.export = { configuration, movie, tv };
```
Comment (pvukmanovic-pannonians) [inline public/media-hunt/API/index.js:5]:
```suggestion
module.exports = { configuration, movie, tv };
```

**Review by pvukmanovic-pannonians — CHANGES_REQUESTED** [review]

File: public/media-hunt/API/index.js (hunk 12 above)
Comment (Ninolina88) [reply public/media-hunt/API/index.js:5]: done

File: public/media-hunt/API/index.js (hunk 11 above)
Comment (Ninolina88) [reply public/media-hunt/API/index.js:3]: done

File: public/media-hunt/API/index.js (hunk 10 above)
Comment (Ninolina88) [reply public/media-hunt/API/index.js:2]: done

File: public/media-hunt/API/index.js (hunk 9 above)
Comment (Ninolina88) [reply public/media-hunt/API/index.js:1]: done

File: public/media-hunt/API/configuration.js (hunk 8 above)
Comment (Ninolina88) [reply public/media-hunt/API/configuration.js:1]: done

File: public/media-hunt/API/tv.js (hunk 7 above)
Comment (Ninolina88) [reply public/media-hunt/API/tv.js:1]: done

File: public/media-hunt/API/movie.js (hunk 6 above)
Comment (Ninolina88) [reply public/media-hunt/API/movie.js:1]: done

**Comment by Ninolina88** [comment]
nije problem resiti index.js konflikt, medjutim, kada sam pulovala promene sa mastera, index.js vristi, jer return nije u funkciji. Nisam sigurna sta je trebalo sa tim delom da se uradi.
Screenshot from 2022-09-14 12-03-06

File: public/media-hunt/.env.example [hunk 13]
```diff
@@ -0,0 +1 @@
+TMDB_API_TOKEN=eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiJmMzEwZDg2NTM4OTkzYmNhMjNlNWUyOWNkYjk3NGFlNCIsInN1YiI6IjVjMWM5OTQwOTI1MTQxMjdmODRlMWVmNSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.csMyKzkblGa8r-2EPryAbFrm_k5LcOZae1tgvTAXxFY
```
Comment (pvukmanovic-pannonians) [inline public/media-hunt/.env.example:1]: Please remove value from example, only the key needs to be commited.

File: public/media-hunt/index.js [hunk 14]
```diff
@@ -1 +1,27 @@
-const app = require("./mediahunt.js");
```
Comment (pvukmanovic-pannonians) [inline public/media-hunt/index.js]: Please revert changes on this file, or rebase from master to fix conflict.

**Review by pvukmanovic-pannonians — CHANGES_REQUESTED** [review]

File: public/media-hunt/.env.example (hunk 13 above)
Comment (Ninolina88) [reply public/media-hunt/.env.example]: removed

File: public/media-hunt/index.js (hunk 14 above)
Comment (Ninolina88) [reply public/media-hunt/index.js]: resolved

**Review by pvukmanovic-pannonians — APPROVED** [review]
LGTM
good job

**Pull request merged**